NxVulnerabilityDetails redesign RSC-232

glenhunter opened the pull request:
https://issues.sonatype.org/browse/RSC-232

Changes to match current Zeplin mockup. Note that some changes aren't possible because of the markup we get from the research group.

`.nx-h3`'s bottom margin was increased to 16px at the request of @dturner32, this is a universal change, not limited to this component.

rpokorny commented on gallery/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sonatype/react-shared-components-gallery",
-  "version": "1.5.0",
+  "version": "1.6.0",
```
Why isn't this a patch version?

glenhunter replied:
> Why isn't this a patch version?

Changed to 1.5.1 https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/e3c6a14749029e9cbb246b38cc66cd4dc7994451

rpokorny commented:
First header should be even with the vertical dividing line, but isn't. 

Zeplin: 
![image](https://user-images.githubusercontent.com/3630091/92491466-44e38e00-f1c0-11ea-99e5-04a5fd762a28.png)

Actual: 
![image](https://user-images.githubusercontent.com/3630091/92491537-56c53100-f1c0-11ea-92f5-b0576866d382.png)

rpokorny commented:
Space between vertical line and text should be 48px, but is about 30px.

Zeplin: 
![image](https://user-images.githubusercontent.com/3630091/92491747-94c25500-f1c0-11ea-91db-bef43c32905f.png)

Actual: 
![image](https://user-images.githubusercontent.com/3630091/92491804-a99ee880-f1c0-11ea-8eff-c25a568ce2d3.png)

rpokorny commented:
Distance from top header to top of vertical divider should be 24, is actually 32.

![image](https://user-images.githubusercontent.com/3630091/92492017-e4a11c00-f1c0-11ea-903e-60fb84a44fdc.png)

rpokorny commented:
The fact that the top margin from the `h2` collapses outside of the top border of the component is concerning.

rpokorny commented:
Distance from h3 headers to following content is 12, should be 16.

rpokorny commented:
Link text color is #00A1E2; should be #45dd1.  I recommend using the `.nx-text-link` class

rpokorny commented:
The zeplin mockup does not demonstrate two paragraphs in succession anywhere, so I can't know whether the spacing there is correct.  @dturner32

rpokorny commented:
The distance from the link text to the link icon seems inconsistent in the zeplin mockup (seems larger on the Weakness line but maybe it's just the font). At any rate, the spacing in the implemented example also seems inconsistent.  Looks like in some places there's margin involved and in some there isn't, and it looks like text spaces are involved in all parts, when they shouldn't be.

rpokorny commented:
Blockquotes should have 8px left border and 24px left padding, but have 5 and 32 respectively.

rpokorny commented:
The elements in the mockup that are impossible given the nature of the markdown (e.g. having the "Reference Link" and the following note be semantically distinct) should be removed from the mockup. @dturner32

rpokorny commented:
Mockup is missing "code block" styles. @dturner32

rpokorny commented:
Malicious code warning should be #cd0028 but is #c70000

rpokorny commented on lib/src/base-styles/_nx-typography.scss:
```diff
@@ -65,7 +65,7 @@
   color: $nx-heading-color;
   font-size: $nx-font-size-heading-level-3;
-  margin-bottom: $nx-spacing-xs;
+  margin-bottom: $nx-spacing-sm;
```
This seems to be the reason for https://github.com/sonatype/sonatype-react-shared-components/pull/129#issuecomment-688931254

glenhunter replied:
> This seems to be the reason for [#129 (comment)](https://github.com/sonatype/sonatype-react-shared-components/pull/129#issuecomment-688931254)

Fixed here: https://github.com/sonatype/sonatype-react-shared-components/commit/fe5d9bf5020868de4de1d8ea122982783c885d85

rpokorny commented:
Zeplin mockup has no space between `<dt>` and `<dd>` elements (ie next to the colons in this screenshot.  That seems like a mistake.

![image](https://user-images.githubusercontent.com/3630091/92494658-f89a4d00-f1c3-11ea-94ba-65c121b14baa.png)

rpokorny commented:
The main text should all be #474747 but is #2c2c2c

rpokorny commented on lib/src/components/NxVulnerabilityDetails/NxVulnerabilityDetails.scss:
```diff
@@ -131,13 +111,32 @@
+      padding: $nx-spacing-l $nx-spacing-l $nx-spacing-l $nx-spacing-xl;
+    }
+
+    // per design make these look like an H4
```
I don't agree with this at all.  We have no guarantee that all `<em>` tags will be used in this way.  When the data researchers write `*stuff*`, their intent is italics.  We should honor that intent not try to guess what deeper meaning they had for that element.

rpokorny replied:
> I don't agree with this at all. We have no guarantee that all <em> tags will be used in this way. When the data researchers write *stuff*, their intent is italics. We should honor that intent not try to guess what deeper meaning they had for that element.

I also no longer see the relevant elements in the mockup.

rpokorny replied:
> I also no longer see the relevant elements in the mockup.

Additionally, I think the content of our example here was misinterpreted.  `*Component Specific Explanation.*` was not something that was present and intended as a header in the original JSON data from the data team.  Rather, it was something _we_ added to the example to give it some "component-specific" content.  "Component Specific Explanation." is not meant to be taken literally as a header here; it is intended to be a placeholder for some actual component-specific explanatory text, which may or may not include `*`s (probably not) in a real example from the data team.

glenhunter replied:
> I don't agree with this at all. We have no guarantee that all `<em>` tags will be used in this way. When the data researchers write `*stuff*`, their intent is italics. We should honor that intent not try to guess what deeper meaning they had for that element.

Ok, restored here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/06b50da56a3aaf5844f3cffe27527768d7c4bc24

rpokorny commented on lib/src/components/NxVulnerabilityDetails/details/DetailLink.tsx:
```diff
@@ -7,13 +7,19 @@
 
 const DetailLink: FunctionComponent<Props> = function DetailLink({href, children}) {
-  return <a href={href} target="_blank" rel="noreferrer">{children}</a>;
+  return <a href={href} target="_blank" rel="noreferrer">
+           {children} <NxFontAwesomeIcon icon={faExternalLinkAlt}/>
+         </a>;
```
Minor: this is how we usually format this:
```suggestion
  return (
    <a href={href} target="_blank" rel="noreferrer">
      {children} <NxFontAwesomeIcon icon={faExternalLinkAlt}/>
    </a>
  );
```

rpokorny commented:
I can't see it in the CSS properties that zeplin displays, but the word "Warning" in the malicious code warning appears to have a heavier font weight than the rest of the text in that element: 
![image](https://user-images.githubusercontent.com/3630091/92497286-259c2f00-f1c7-11ea-883e-74a822d38be3.png)

rpokorny commented on lib/src/components/NxVulnerabilityDetails/details/MarkdownDetail.tsx:
```diff
@@ -22,7 +24,9 @@ const MarkdownDetail: FunctionComponent<Props> =
     return markdown || componentSpecificMarkdown ? (
       <RenderDetail title={title}>
         {warning &&
-          <p className="nx-vulnerability-details__malicious-code-warning"><strong>{warning}</strong></p>
```
I'd argue that regardless of styling, `<strong>` is the right semantics for this content.

glenhunter replied:
> I'd argue that regardless of styling, `<strong>` is the right semantics for this content.

Restored here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/700622f5b6c6cdc35917f03dc33e6ba1e3ee66fd

glenhunter commented:
> I can't see it in the CSS properties that zeplin displays, but the word "Warning" in the malicious code warning appears to have a heavier font weight than the rest of the text in that element:
> ![image](https://user-images.githubusercontent.com/3630091/92497286-259c2f00-f1c7-11ea-883e-74a822d38be3.png)

That seems to be a string we get from the backend so I don't see any way we could uniquely style it. @dturner32

glenhunter commented:
> Zeplin mockup has no space between `<dt>` and `<dd>` elements (ie next to the colons in this screenshot). That seems like a mistake.
> 
> ![image](https://user-images.githubusercontent.com/3630091/92494658-f89a4d00-f1c3-11ea-94ba-65c121b14baa.png)

For @dturner32 's reference the current spacing is `8px`

glenhunter commented:
> The main text should all be #474747 but is #2c2c2c
> 
> Edit: I see now you have the correct text color on some `<p>`s but not others nor on anything in the left column

I am a bit curious about why this wouldn't be the default font colour. @dturner32 ?

glenhunter commented:
> Malicious code warning should be #cd0028 but is #c70000.
> 
> Edit: same issue with the inline code snippets.

Fixed here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/e39f41e6888a2a7713db1851193a4bfb6d2a8951

and I assumed that we'd want the code block text colour to be universal so I change `%nx-code`

rpokorny commented:
> That seems to be a string we get from the backend so I don't see any way we could uniquely style it. @dturner32

That string is defined in `NxVulnerabilityDetails.tsx`; it's not from the backend.

glenhunter commented:
> Space between vertical line and text should be 48px, but is about 30px.
> 
> Zeplin:
> ![image](https://user-images.githubusercontent.com/3630091/92491747-94c25500-f1c0-11ea-91db-bef43c32905f.png)
> 
> Actual:
> ![image](https://user-images.githubusercontent.com/3630091/92491804-a99ee880-f1c0-11ea-8eff-c25a568ce2d3.png)

https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/5a2859aea645426262b4ac126253f67ac39821d3

Also a note for @dturner32 our spacing variables jump from `$nx-spacing-xl: 32px` to `$nx-spacing-xxl: 64px` and if `48px` is going to show up more often we might want to adjust that

glenhunter commented:
> First header should be even with the vertical dividing line, but isn't.
> 
> Zeplin:
> ![image](https://user-images.githubusercontent.com/3630091/92491466-44e38e00-f1c0-11ea-99e5-04a5fd762a28.png)
> 
> Actual:
> ![image](https://user-images.githubusercontent.com/3630091/92491537-56c53100-f1c0-11ea-92f5-b0576866d382.png)
> 
> Edit: similar issue with the bottom of the component.

Fixed top and bottom: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/aa53cd43861d41243d6ae199be1b23262887bb84

glenhunter commented:
> The fact that the top margin from the `h2` collapses outside of the top border of the component is concerning.

https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/5862cfeede9350f14d9c8d1874b13e8a4a58eb50

glenhunter commented:
> Distance from h3 headers to following content is 12, should be 16.

I goofed and used the wrong variable, fixed here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/fe5d9bf5020868de4de1d8ea122982783c885d85

Per discussion with Derrek this is a universal change to `H3`'s

dturner32 commented:
> The zeplin mockup does not demonstrate two paragraphs in succession anywhere, so I can't know whether the spacing there is correct. @dturner32

Padding between 2 or more paragraphs should be 24px

dturner32 commented:
> The distance from the link text to the link icon seems inconsistent in the zeplin mockup (seems larger on the Weakness line but maybe it's just the font). At any rate, the spacing in the implemented example also seems inconsistent. Looks like in some places there's margin involved and in some there isn't, and it looks like text spaces are involved in all parts, when they shouldn't be.

Just double-checked this and it's a font thing. There's a small amount of space that the bounding box adds after the number 1 that I don't really have any control over. The padding between the link text and the icon should be 8px though

dturner32 commented:
> The elements in the mockup that are impossible given the nature of the markdown (e.g. having the "Reference Link" and the following note be semantically distinct) should be removed from the mockup. @dturner32

just updated this in Zeplin

dturner32 commented:
> Mockup is missing "code block" styles. @dturner32

updated in Zeplin

dturner32 commented:
> I can't see it in the CSS properties that zeplin displays, but the word "Warning" in the malicious code warning appears to have a heavier font weight than the rest of the text in that element:
> ![image](https://user-images.githubusercontent.com/3630091/92497286-259c2f00-f1c7-11ea-883e-74a822d38be3.png)

'Warning' is in semibold

dturner32 commented:
> > I can't see it in the CSS properties that zeplin displays, but the word "Warning" in the malicious code warning appears to have a heavier font weight than the rest of the text in that element:
> > ![image](https://user-images.githubusercontent.com/3630091/92497286-259c2f00-f1c7-11ea-883e-74a822d38be3.png)
> 
> That seems to be a string we get from the backend so I don't see any way we could uniquely style it. @dturner32

if there is no way to style it then the whole thing should be regular weight

dturner32 commented:
> > The main text should all be #474747 but is #2c2c2c
> > Edit: I see now you have the correct text color on some `<p>`s but not others nor on anything in the left column
> 
> I am a bit curious about why this wouldn't be the default font colour. @dturner32 ?

all the main text should be #474747. Just checked in Zeplin and that's what it is showing. Only the headers should be #2c2c2c

glenhunter commented:
> The distance from the link text to the link icon seems inconsistent in the zeplin mockup (seems larger on the Weakness line but maybe it's just the font). At any rate, the spacing in the implemented example also seems inconsistent. Looks like in some places there's margin involved and in some there isn't, and it looks like text spaces are involved in all parts, when they shouldn't be.

https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/83d4d73caa3ed8a50d2dd3ea401ddfd320cbdd02

glenhunter commented:
> > > The main text should all be #474747 but is #2c2c2c
> > > Edit: I see now you have the correct text color on some `<p>`s but not others nor on anything in the left column
> > 
> > 
> > I am a bit curious about why this wouldn't be the default font colour. @dturner32 ?
> 
> all the main text should be #474747. Just checked in Zeplin and that's what it is showing. Only the headers should be #2c2c2c

Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/bb996704c0d2b1a411b895fd0b779c07427f6894

rpokorny commented:
> Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: bb99670

I feel like this thread might be hinting at a bigger miscommunication.  @dturner32 , what color is content text (e.g. paragraph text) supposed to be in general (as opposed to particularly in this component)?

dturner32 commented:
> > Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: [bb99670](https://github.com/sonatype/sonatype-react-shared-components/commit/bb996704c0d2b1a411b895fd0b779c07427f6894)
> 
> I feel like this thread might be hinting at a bigger miscommunication. @dturner32 , what color is content text (e.g. paragraph text) supposed to be in general (as opposed to particularly in this component)?

content text should be #474747

glenhunter commented:
> > > Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: [bb99670](https://github.com/sonatype/sonatype-react-shared-components/commit/bb996704c0d2b1a411b895fd0b779c07427f6894)
> > 
> > 
> > I feel like this thread might be hinting at a bigger miscommunication. @dturner32 , what color is content text (e.g. paragraph text) supposed to be in general (as opposed to particularly in this component)?
> 
> content text should be #474747

Right now #2c2c2c is our default text colour applied to the `<body>` tag, but this sounds like it should be #474747, with #2c2c2c representing a "dark" variant (applied to headings) and perhaps also having a lighter variant as well

rpokorny commented:
> > Space between vertical line and text should be 48px, but is about 30px.
> > Zeplin:
> > ![image](https://user-images.githubusercontent.com/3630091/92491747-94c25500-f1c0-11ea-91db-bef43c32905f.png)
> > Actual:
> > ![image](https://user-images.githubusercontent.com/3630091/92491804-a99ee880-f1c0-11ea-8eff-c25a568ce2d3.png)
> 
> [5a2859a](https://github.com/sonatype/sonatype-react-shared-components/commit/5a2859aea645426262b4ac126253f67ac39821d3)
> 
> Also a note for @dturner32 our spacing variables jump from `$nx-spacing-xl: 32px` to `$nx-spacing-xxl: 64px` and if `48px` is going to show up more often we might want to adjust that

Given how weird that fix is, I want to ask why this view deviates from the normal `nx-grid` spacing.  Given that we haven't touched on `nx-grid` in a long time, is it's current spacing even considered correct? @dturner32

rpokorny commented:
> > The distance from the link text to the link icon seems inconsistent in the zeplin mockup (seems larger on the Weakness line but maybe it's just the font). At any rate, the spacing in the implemented example also seems inconsistent. Looks like in some places there's margin involved and in some there isn't, and it looks like text spaces are involved in all parts, when they shouldn't be.
> 
> [83d4d73](https://github.com/sonatype/sonatype-react-shared-components/commit/83d4d73caa3ed8a50d2dd3ea401ddfd320cbdd02)

the `<a>` should use `container-horizontal` to avoid the space between the icon and the following comma: 
![image](https://user-images.githubusercontent.com/3630091/92770466-fa047a80-f367-11ea-88d1-ac05f08fd8bf.png)

glenhunter commented:
> > > The distance from the link text to the link icon seems inconsistent in the zeplin mockup (seems larger on the Weakness line but maybe it's just the font). At any rate, the spacing in the implemented example also seems inconsistent. Looks like in some places there's margin involved and in some there isn't, and it looks like text spaces are involved in all parts, when they shouldn't be.
> > 
> > 
> > [83d4d73](https://github.com/sonatype/sonatype-react-shared-components/commit/83d4d73caa3ed8a50d2dd3ea401ddfd320cbdd02)
> 
> the `<a>` should use `container-horizontal` to avoid the space between the icon and the following comma:
> ![image](https://user-images.githubusercontent.com/3630091/92770466-fa047a80-f367-11ea-88d1-ac05f08fd8bf.png)

This also required wrapping the link text in a `<span>` to get rid of the bare text node problem: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/aa7a80879a59aaa476fe5d9197d0ac220be44adc

dturner32 commented:
> > > Space between vertical line and text should be 48px, but is about 30px.
> > > Zeplin:
> > > ![image](https://user-images.githubusercontent.com/3630091/92491747-94c25500-f1c0-11ea-91db-bef43c32905f.png)
> > > Actual:
> > > ![image](https://user-images.githubusercontent.com/3630091/92491804-a99ee880-f1c0-11ea-8eff-c25a568ce2d3.png)
> > 
> > 
> > [5a2859a](https://github.com/sonatype/sonatype-react-shared-components/commit/5a2859aea645426262b4ac126253f67ac39821d3)
> > Also a note for @dturner32 our spacing variables jump from `$nx-spacing-xl: 32px` to `$nx-spacing-xxl: 64px` and if `48px` is going to show up more often we might want to adjust that
> 
> Given how weird that fix is, I want to ask why this view deviates from the normal `nx-grid` spacing. Given that we haven't touched on `nx-grid` in a long time, is it's current spacing even considered correct? @dturner32

after looking at nx-grid in the RSC library I'd want revisit the overall layout. The short answer is no, the current spacing in nx-grid technically wouldn't be correct given the way things are set up in the vulnerability details right now.

dturner32 commented:
> > > > Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: [bb99670](https://github.com/sonatype/sonatype-react-shared-components/commit/bb996704c0d2b1a411b895fd0b779c07427f6894)
> > > 
> > > 
> > > I feel like this thread might be hinting at a bigger miscommunication. @dturner32 , what color is content text (e.g. paragraph text) supposed to be in general (as opposed to particularly in this component)?
> > 
> > 
> > content text should be #474747
> 
> Right now #2c2c2c is our default text colour applied to the `<body>` tag, but this sounds like it should be #474747, with #2c2c2c representing a "dark" variant (applied to headings) and perhaps also having a lighter variant as well

correct, #474747 is designated for body content and #2c2c2c applies to all headers, sub headers and labels. In other words, any H tags should appear in #2c2c2c and anything that isn't an H tag is #474747

glenhunter commented:
> Link text color is #00A1E2; should be #45dd1. I recommend using the `.nx-text-link` class

It is but I cannot find the commit, I wonder if it was fixed elsewhere and came in via a merge from master

glenhunter commented:
> Blockquotes should have 8px left border and 24px left padding, but have 5 and 32 respectively.

Fixed here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/a2eef94e67357a407cf62035fcfc8282be04ff8b

glenhunter commented:
> > The zeplin mockup does not demonstrate two paragraphs in succession anywhere, so I can't know whether the spacing there is correct. @dturner32
> 
> Padding between 2 or more paragraphs should be 24px

Adjusted here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/e7ca0a18c697f1f53e665fc178bd1316506b5bd5

glenhunter commented:
> > > > > Ok, that didn't really answer my question about why we'd want to have all that text use the light variation, but it's changed here: [bb99670](https://github.com/sonatype/sonatype-react-shared-components/commit/bb996704c0d2b1a411b895fd0b779c07427f6894)
> > > > 
> > > > 
> > > > I feel like this thread might be hinting at a bigger miscommunication. @dturner32 , what color is content text (e.g. paragraph text) supposed to be in general (as opposed to particularly in this component)?
> > > 
> > > 
> > > content text should be #474747
> > 
> > 
> > Right now #2c2c2c is our default text colour applied to the `<body>` tag, but this sounds like it should be #474747, with #2c2c2c representing a "dark" variant (applied to headings) and perhaps also having a lighter variant as well
> 
> correct, #474747 is designated for body content and #2c2c2c applies to all headers, sub headers and labels. In other words, any H tags should appear in #2c2c2c and anything that isn't an H tag is #474747

I have created a ticket to address this since I think the change is much too large to include in this PR: https://issues.sonatype.org/browse/RSC-276

glenhunter commented:
> > > Space between vertical line and text should be 48px, but is about 30px.
> > > Zeplin:
> > > ![image](https://user-images.githubusercontent.com/3630091/92491747-94c25500-f1c0-11ea-91db-bef43c32905f.png)
> > > Actual:
> > > ![image](https://user-images.githubusercontent.com/3630091/92491804-a99ee880-f1c0-11ea-8eff-c25a568ce2d3.png)
> > 
> > 
> > [5a2859a](https://github.com/sonatype/sonatype-react-shared-components/commit/5a2859aea645426262b4ac126253f67ac39821d3)
> > Also a note for @dturner32 our spacing variables jump from `$nx-spacing-xl: 32px` to `$nx-spacing-xxl: 64px` and if `48px` is going to show up more often we might want to adjust that
> 
> Given how weird that fix is, I want to ask why this view deviates from the normal `nx-grid` spacing. Given that we haven't touched on `nx-grid` in a long time, is it's current spacing even considered correct? @dturner32

As discussed moved the whole of the padding to a custom class on the grid column https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/9e7484d3a994554651a16ff73b57ecd9a3b2e389

glenhunter commented:
> > Link text color is #00A1E2; should be #45dd1. I recommend using the `.nx-text-link` class
> 
> It is but I cannot find the commit, I wonder if it was fixed elsewhere and came in via a merge from master

Correct link class employed here: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/f9fc5e4526a9efa1be71eb48531cafab222a499b

dturner32 commented:
> > Zeplin mockup has no space between `<dt>` and `<dd>` elements (ie next to the colons in this screenshot). That seems like a mistake.
> > ![image](https://user-images.githubusercontent.com/3630091/92494658-f89a4d00-f1c3-11ea-94ba-65c121b14baa.png)
> 
> For @dturner32 's reference the current spacing is `8px`

updated in Zeplin

rpokorny commented:
> Correct link class employed here: f9fc5e4

The custom style with the wrong color still remains and is taking precedence.

rpokorny commented:
> > correct, #474747 is designated for body content and #2c2c2c applies to all headers, sub headers and labels. In other words, any H tags should appear in #2c2c2c and anything that isn't an H tag is #474747
> 
> I have created a ticket to address this since I think the change is much too large to include in this PR: https://issues.sonatype.org/browse/RSC-276

In that case I think the custom style for it that was added in this PR should be removed.

rpokorny commented:
> > Padding between 2 or more paragraphs should be 24px
> 
> Adjusted here: e7ca0a1

That didn't work because the paragraphs in the vulnerability details don't have `.nx-p`

glenhunter commented:
> > > Padding between 2 or more paragraphs should be 24px
> > 
> > 
> > Adjusted here: [e7ca0a1](https://github.com/sonatype/sonatype-react-shared-components/commit/e7ca0a18c697f1f53e665fc178bd1316506b5bd5)
> 
> That didn't work because the paragraphs in the vulnerability details don't have `.nx-p`

Right: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/4f42d20c6f62d647d9e821c3cdf6ef6fe9e3a0f5 

The custom `.nx-p` style is also removed. Per Derrek that should be 24px for all `<p>` tags across the board, but that's a very broad change to put into this PR

glenhunter commented:
> > > correct, #474747 is designated for body content and #2c2c2c applies to all headers, sub headers and labels. In other words, any H tags should appear in #2c2c2c and anything that isn't an H tag is #474747
> > 
> > 
> > I have created a ticket to address this since I think the change is much too large to include in this PR: https://issues.sonatype.org/browse/RSC-276
> 
> In that case I think the custom style for it that was added in this PR should be removed.

https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/523945b1b7b3222c309ec966d74acef798127e2b

glenhunter commented:
> > > > Padding between 2 or more paragraphs should be 24px
> > > 
> > > 
> > > Adjusted here: [e7ca0a1](https://github.com/sonatype/sonatype-react-shared-components/commit/e7ca0a18c697f1f53e665fc178bd1316506b5bd5)
> > 
> > 
> > That didn't work because the paragraphs in the vulnerability details don't have `.nx-p`
> 
> Right: [4f42d20](https://github.com/sonatype/sonatype-react-shared-components/commit/4f42d20c6f62d647d9e821c3cdf6ef6fe9e3a0f5)
> 
> The custom `.nx-p` style is also removed. Per Derrek that should be 24px for all `<p>` tags across the board, but that's a very broad change to put into this PR



> > Correct link class employed here: [f9fc5e4](https://github.com/sonatype/sonatype-react-shared-components/commit/f9fc5e4526a9efa1be71eb48531cafab222a499b)
> 
> The custom style with the wrong color still remains and is taking precedence.

Removed all of the custom text link styling: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/2f5c3dc0aaabf912138a845f309f6426453bd5b1

rpokorny commented:
> Removed all of the custom text link styling: 2f5c3dc

Well now they have underlines.  According to Zeplin, they shouldn't.

glenhunter commented:
> > Removed all of the custom text link styling: [2f5c3dc](https://github.com/sonatype/sonatype-react-shared-components/commit/2f5c3dc0aaabf912138a845f309f6426453bd5b1)
> 
> Well now they have underlines. According to Zeplin, they shouldn't.

https://issues.sonatype.org/browse/RSC-280

rpokorny commented on lib/src/components/NxVulnerabilityDetails/NxVulnerabilityDetails.scss:
```diff
@@ -131,9 +118,13 @@
+      border-left-width: 8px;
+      border-radius: $nx-border-radius;
+      margin: 0 0 $nx-spacing-l 0;
+      padding: $nx-spacing-l;
```
A lot of duplication between this and `pre`

glenhunter replied:
> A lot of duplication between this and `pre`; could be reduced with a mixin or an `@extends`

Probably not exactly what you were thinking, but I think it's clear: https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/c4bdee8cb4e1e1f8ad243b06cca39aa053fb1ec1

rpokorny replied:
> > A lot of duplication between this and pre; could be reduced with a mixin or an @extends
>
> Probably not exactly what you were thinking, but I think it's clear: c4bdee8

That doesn't reduce the duplication, the styles are still present twice (now in `blockquote` and in `%nx-pre`)

glenhunter replied:
> > > A lot of duplication between this and pre; could be reduced with a mixin or an @extends
> > 
> > 
> > Probably not exactly what you were thinking, but I think it's clear: [c4bdee8](https://github.com/sonatype/sonatype-react-shared-components/commit/c4bdee8cb4e1e1f8ad243b06cca39aa053fb1ec1)
> 
> That doesn't reduce the duplication, the styles are still present twice (now in `blockquote` and in `%nx-pre`)

I really don't mind the duplication between `pre` and `blockquote` they are very different elements. If the thick left border style starts popping up more frequently then we could make up a mixin for it but right now we have just these two instances. I see no point in having them share spacing values (in an extend or a mixin) there's too much risk of a designer changing those.

rpokorny commented on lib/src/components/NxVulnerabilityDetails/__tests__/DetailLink.test.tsx:
```diff
@@ -23,6 +23,6 @@ describe('DetailLink', function() {
     expect(shallowRender).toHaveProp('rel', 'noreferrer');
-    expect(shallowRender).toHaveText('Label');
+    expect(shallowRender).toHaveText('Label<NxFontAwesomeIcon />');
   });
```
Should really have an additional test checking what kind of icon it is

glenhunter replied:
> Should really have an additional test checking what kind of icon it is

https://github.com/sonatype/sonatype-react-shared-components/pull/129/commits/4954fcc98780510b3dd1bcd8e74d5a018bbdade3
